python/cudf/cudf/core/frame.py

```diff
@@ -3812,3 +3813,37 @@ def _is_series(obj):
     instead of checking for isinstance(obj, cudf.Series)
     """
     return isinstance(obj, Frame) and obj.ndim == 1 and obj._index is not None
+
+
+def _drop_rows_by_labels(
+    obj: Union[cudf.DataFrame, cudf.Series],
```

Comment:
I think there's a `DataFrameOrSeries` type in `_typing.py` that we can use instead.